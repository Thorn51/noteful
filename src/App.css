/* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/

html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}
body {
  line-height: 1;
}
ol,
ul {
  list-style: none;
}
blockquote,
q {
  quotes: none;
}
blockquote:before,
blockquote:after,
q:before,
q:after {
  content: "";
  content: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}

/* Header */

.App_header {
  background: grey;
  color: white;
  justify-content: center;
  height: 15vh;
}

.App_title {
  height: 100%;
  text-align: center;
  font-size: 3rem;
  font-weight: bolder;
  line-height: 9rem;
}

.App_title,
a {
  text-decoration: none;
  color: white;
}

/* App Main */

.app_main {
  display: flex;
  flex-direction: row;
}

.folder_sidebar {
  flex: 1;
  padding: 15px;
}

.notes_section {
  flex: 3;
  padding: 15px;
}
