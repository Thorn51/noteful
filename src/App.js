import React from "react";
import { Route, Link } from "react-router-dom";
import "./App.css";
import Sidebar from "./Sidebar/Sidebar";
import Main from "./Main/Main";
import Folder from "./Folder/Folder";
import NoteSidebar from "./NoteSidebar/NoteSidebar";
import NoteMain from "./NoteMain/NoteMain";
import NoteContext from "./NoteContext";
import AddFolder from "./AddFolder/AddFolder";
import AddNote from "./AddNote/AddNote";
import MainError from "./ErrorBoundaries/MainError";

export default class App extends React.Component {
  state = {
    folders: [],
    notes: []
  };

  addFolder = folder => {
    this.setState({
      folders: [...this.state.folders, folder]
    });
  };

  addNote = note => {
    this.setState({
      notes: [...this.state.notes, note]
    });
  };

  deleteNote = noteId => {
    const updateNotes = this.state.notes.filter(note => note.id !== noteId);
    const options = {
      method: "DELETE"
    };

    console.log(noteId);

    fetch("http://localhost:8000/api/notes/" + noteId, options)
      .then(response => {
        if (!response.ok) {
          throw new Error("There has been a problem removing the note");
        } else {
          return response.json();
        }
      })
      .then(data => {
        this.setState({
          notes: updateNotes
        });
      })
      .catch(error => {
        console.log(error);
      });
  };

  componentDidMount() {
    fetch("http://localhost:8000/api/folders")
      .then(response => {
        if (!response.ok) {
          throw new Error(
            "There has been an error fetching the folders from local host 9090"
          );
        } else {
          return response.json();
        }
      })
      .then(data => {
        this.setState({
          folders: data
        });
      })
      .catch(error => {
        console.log(error);
      });

    fetch("http://localhost:8000/api/notes")
      .then(response => {
        if (!response.ok) {
          throw new Error(
            "There has been an error fetching the notes from local host 9090"
          );
        } else {
          return response.json();
        }
      })
      .then(data => {
        this.setState({
          notes: data
        });
      })
      .catch(error => {
        console.log(error);
      });
  }
  render() {
    const contextValue = {
      folders: this.state.folders,
      notes: this.state.notes,
      addFolder: this.addFolder,
      addNote: this.addNote,
      deleteNote: this.deleteNote
    };

    return (
      <div>
        <header className="App_header">
          <Link to="/">
            <h1 className="App_title">Noteful</h1>
          </Link>
        </header>
        <main className="app_main">
          <NoteContext.Provider value={contextValue}>
            {/* Sidebar */}
            <Route exact path="/" component={Sidebar} />
            <Route path="/folder" component={Sidebar} />
            <Route path="/note/:noteId" component={NoteSidebar} />
            <Route path="/add-folder" component={AddFolder} />

            {/* Main */}
            <MainError>
              <Route exact path="/" component={Main} />
              <Route path="/folder/:folderId" component={Folder} />
              <Route path="/note/:noteId" component={NoteMain} />
              <Route path="/add-note" component={AddNote} />
            </MainError>
          </NoteContext.Provider>
        </main>
      </div>
    );
  }
}
