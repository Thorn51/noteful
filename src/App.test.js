import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

describe("App", () => {
  it("renders without throwing errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the component as expected", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
