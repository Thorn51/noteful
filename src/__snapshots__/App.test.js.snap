// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component as expected 1`] = `
<div>
  <header
    className="App_header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="App_title"
      >
        Noteful
      </h1>
    </a>
  </header>
  <main
    className="app_main"
  >
    <aside
      className="folder_sidebar"
    >
      <section
        className="folder_list"
      />
      <a
        href="/add-folder"
        onClick={[Function]}
      >
        <div
          className="add_folder"
        >
          Add Folder
        </div>
      </a>
    </aside>
    <section
      className="notes_section"
    >
      <div
        className="note_list"
      />
      <a
        href="/add-note"
        onClick={[Function]}
      >
        <button
          className="add_note"
        >
          Add Note
        </button>
      </a>
    </section>
  </main>
</div>
`;
